src/import/load-costume.js

```diff
@@ -82,21 +52,85 @@ const loadCostumeFromAsset = function (costume, costumeAsset, runtime, optVersio
         };
         imageElement.addEventListener('error', onError);
         imageElement.addEventListener('load', onLoad);
-        imageElement.src = costumeAsset.encodeDataURI();
+        let src = costumeAsset.encodeDataURI();
+        if (costume.bitmapResolution === 1 && !runtime.v2BitmapAdapter) {
+            log.error('No V2 bitmap adapter present; bitmaps may not render correctly.');
+        } else if (costume.bitmapResolution === 1) {
+            src = runtime.v2BitmapAdapter.convertResolution1Bitmap(src, dataURI => {
+                if (dataURI) {
+                    // Put back into storage
+                    const storage = runtime.storage;
+                    costume.assetId = storage.builtinHelper.cache(
+                        storage.AssetType.ImageBitmap,
+                        storage.DataFormat.PNG,
+                        runtime.v2BitmapAdapter.convertDataURIToBinary(dataURI)
+                    );
+                    costume.md5 = `${costume.assetId}.${costume.dataFormat}`;
+                    if (rotationCenter) {
+                        rotationCenter[0] = rotationCenter[0] * 2;
+                        rotationCenter[1] = rotationCenter[1] * 2;
+                        costume.rotationCenterX = rotationCenter[0];
+                        costume.rotationCenterY = rotationCenter[1];
+                    }
+                    costume.bitmapResolution = 2;
```

Comment:
So if an error occurs during the conversion process, we don't end up updating the bitmapResolution, is that any cause for concern?